bin/gen_static.jl

```diff
@@ -43,25 +28,11 @@ function make_tarball(
     tarball::AbstractString,
     tree_path::AbstractString,
 )
-    paths = String[]
-    for (root, dirs, files) in walkdir(tree_path)
-        path = root != tree_path ? relpath(root, tree_path) : ""
-        for file in [dirs; files]
-            push!(paths, joinpath(path, file))
+    open(tarball, write=true) do io
+        open(pipeline(compress, io), write=true) do io
+            Tar.create(tree_path, io)
```

Comment:
I've observed a performance regression forking this PR and #32 when building tarballs for General registry and `FFTW_jll@3.3.9+5`:

* before: 24.134260 seconds (7.35 M allocations: 472.674 MiB, 1.60% gc time)
* after: 32.119186 seconds (5.74 M allocations: 438.242 MiB, 0.39% gc time)

`Juno.@profiler` tells `Tar.create` is the bottleneck.

The overall changes can be found at https://github.com/johnnychen94/StorageServer.jl/pull/1/commits/f77fb30e8700a72cf882a825df99f70ef546a0ad

---

```bash
bash-3.2$ gtar --version
tar (GNU tar) 1.30
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
```

```julia
julia> versioninfo()
Julia Version 1.4.1
Commit 381693d3df* (2020-04-14 17:20 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.7.0)
  CPU: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-8.0.1 (ORCJIT, skylake)
Environment:
  JULIA_NUM_THREADS = 8

(StorageServer) pkg> st Tar
Project StorageServer v0.1.0
Status `~/Documents/Julia/StorageServer/Project.toml`
  [a4e569a6] Tar v1.3.0
```

Reply:
Being 33% slower than the hyper-optimized GNU tar is quite good. I've created an issue about using sendfile to optimize tarball creation and extraction: https://github.com/JuliaIO/Tar.jl/issues/33. However, matching performance of GNU tar isn't really a high priority. It's also possible since we're not sending data directly to a real file descriptor but to a `TranscodingStream` that the sendfile optimization wouldn't actually help in this case, but it's also possible that we should be using a buffer size that matches what `TranscodingStreams` uses and maybe avoid multiple buffers. In general, what's needed is an API for saying "send this much data from here to here" that just does whatever is most efficient for the given source and destination.